src/include/buffer.h

```diff
@@ -297,11 +309,11 @@ class CEPH_BUFFER_API buffer {
 
       void advance(int o);
       void seek(unsigned o);
-      char operator*();
-      iterator& operator++();
-      ptr get_current_ptr();
+      char operator*() const;
+      iterator_impl& operator++();
+      ptr get_current_ptr() const;
```

Comment:
So this means that users can't modify individual bytes or buffer pointers any more, right? Does that actually work for us? I thought we overwrote specific bytes in a few places (the ENCODE_START family of macros is what I'm thinking of...but it looks like we use the copy_in function there so I guess it's all good).